portfolio/src/main/webapp/game.js

```diff
@@ -118,33 +118,49 @@ class Board {
         this.board[row][col] = 2;
       }
       this.numTiles++;
-      this.checkGameState(this.redTrue);
+      this.checkGameState();
       this.redTrue = !this.redTrue;
     } else {
         alert("Cannot place tile there. Try again.");
     }
   }
   
+  /*
+  * Shows win message
+  */ 
+  showWinMessage() {
+    if (this.redTrue) {
+      if (confirm("Win for red! Play again?")) {
+            this.init();
+      }
+    } else {
+      if (confirm("Win for black! Play again?")) {
+            this.init();
+      }
+    }  
+  }
+
   /*
   * Checks whether move made by redTrue player resulted in win or if game board is full.
   * Sends alert with win message or gameOver message if board is full and allows user to
   * restart game.  
   */
-  checkGameState(redTrue) {
-    var player = 0;
-    if (redTrue) {
+  checkGameState() {
```

Comment:
Thanks for making this more efficient!

One more nit here :):
The function now has gotten pretty. A good rule of thumb is that the whole body of the function should fit on the screen without having to scroll. I would use helper functions. For example, I would create different function for checkWinRow, checkWinCol, and checkWinDiaganol. Each of these function would return a boolean on if there is a win. Then I would do the following
```suggestion
  checkGameState() {
    .....
    if (checkWinRow() || checkWinCol() || checkWinDiaganol()) {
      this.showWinMessage();
    }
  }
```

Note, we rely on short circuiting to make sure we don't checkCol or checkDiagonal when checkRow returns true.
https://stackoverflow.com/questions/9344305/what-is-short-circuiting-and-how-is-it-used-when-programming-in-java